packages/react-composites/src/composites/ChatComposite/adapter/AzureCommunicationChatAdapter.test.ts

```diff
@@ -29,6 +67,47 @@ describe('Error is reflected in state and events', () => {
     expect(errorListener.errors.length).toBe(1);
     expect(errorListener.errors[0].operation).toBe('ChatThreadClient.updateTopic');
   });
+
+  it('when listMessages fails on iteration', async () => {
+    const threadClient = new StubChatThreadClient();
+    threadClient.listMessages = (): PagedAsyncIterableIterator<any> => {
+      return failingPagedAsyncIterator(new Error('injected error'));
+    };
+    const adapter = await createChatAdapterWithStubs(new StubChatClient(threadClient));
+    const stateListener = new StateChangeListener(adapter);
+    const errorListener = new ErrorListener(adapter);
+
+    await expect(adapter.loadPreviousChatMessages(3)).rejects.toThrow();
+
+    expect(stateListener.onChangeCalledCount).toBe(1);
+    const latestError = stateListener.state.latestErrors['ChatThreadClient.listMessages'];
+    expect(latestError).toBeDefined();
+    expect(errorListener.errors.length).toBe(1);
+    expect(errorListener.errors[0].operation).toBe('ChatThreadClient.listMessages');
+  });
+
+  /*
+  // This test current fails.
+  // If `listMessages` fails immediately (network flake), the adaptor construction throws an exception,
```

Comment:
Is this saying if list lsitMessages fails due to a network issue the composite throws a critical error it can't recover from?

Reply:
Yep, fixed in https://github.com/Azure/communication-ui-library/pull/550

Reply:
Aha! Nice :D